Updating the installation guide (#616)
MicrosoftDocs/quantum-docs-pr: articles/install-guide/pyinstall.md

---
title: Develop with Q# + Python
author: natke
ms.author: nakersha
ms.date: 9/30/2019
ms.topic: article
ms.custom: how-to
uid: microsoft.quantum.install.python
---

# Develop with Q# + Python

Install the QDK to develop Python host programs to call Q# operations.

1. Pre-requisites

    - [Python](https://www.python.org/downloads/) 3.6 or later
    - The [PIP](https://pip.pypa.io/en/stable/installing) Python package manager
    - [.NET Core SDK 3.0 or later](https://www.microsoft.com/net/download)

> 1. Install the `qsharp` package, a Python package that allows the interoperation between Q# and Python.

Reviewer: ```suggestion
1. Install the `qsharp` package, a Python package that enables interop between Q# and Python.
```

    ```bash
    pip install qsharp
    ```

> 1. Install the `iqsharp`, an extension primarily used by Jupyter and Python that provides the core functionality for compiling and simulating Q# operations.

Reviewer: ```suggestion
1. Install `iqsharp`, an extension primarily used by Jupyter and Python that provides the core functionality for compiling and executing Q# operations.
```

Reviewer 2: Strictly speaking, IQ# isn't an extension, but a kernel.

    ```bash
    dotnet tool install -g Microsoft.Quantum.IQSharp
    dotnet iqsharp install
    ```
  

> 1. Although it's not strictly necessary, we highly recommend to use the lightweight Visual Studio Code (VS Code) IDE to edit the Q# and Python files.

Reviewer: ```suggestion
1. While you can use Q# with Python in any IDE, we highly recommend using Visual Studio Code (VS Code) IDE for your Q# and Python applications. By using Visual Studio Code and the QDK Visual Studio Code extension, you gain access to richer functionality like syntax highlighting.
```

    - Install [VS Code](https://code.visualstudio.com/download) (Windows, Linux and Mac)
    - Install the [QDK extension for VS Code](https://marketplace.visualstudio.com/items?itemName=quantum.quantum-devkit-vscode).

1. Verify the installation by creating a `Hello World` application

    - Create a minimal Q# operation, by creating a file called `Operation.qs`, and adding the following code to it:

        ```qsharp
        namespace HelloWorld
        {
            open Microsoft.Quantum.Intrinsic;
            open Microsoft.Quantum.Canon;

            operation SayHello() : Unit {
                Message("Hello from quantum world!");
            }
        }
        ```

    - Create a Python program called `hello_world.py` to call the Q# `SayHello()` operation:

        ```python
        import qsharp

        from HelloWorld import SayHello

        SayHello.simulate()
        ```

    - Run the program:

        ```bash
        python hello_world.py
        ```

    - Verify the output. Your program should output the following lines:

        ```bash
        Hello from quantum world!
       ```

> [!NOTE]

> > * You can also use Python Jupyter notebooks to write the classical Python program and call from the cells the Q# operations. The Python code is just a normal Python program.

Reviewer: ```suggestion
> * You can also use Python Jupyter notebooks to write the classical Python program and call Q# operations from the cells. The Python code is just a normal Python program.
```

---
title: Develop with Q# + Python
author: natke
ms.author: nakersha
ms.date: 9/30/2019
ms.topic: article
ms.custom: how-to
uid: microsoft.quantum.install.python
---

> # Develop with Q# + Python

Reviewer: ```suggestion
# Develop with Q# & Python
```

Author: The '+' and the '&' basically mean the same. However, I like more the '+' because it conveys a "modularity feeling" that the '&' does not. And I think modularity is a good feature of Q#, the same Q# code can be called from C# or from Python. Let me know what you think about this before substituting all the '+' with '&'.

Reviewer: I'd like some input from @natke and/or @cjgronlund on how this is typically done in docs. I think for a heading this is a grammar thing and not a how we position the language thing, but it'd be nice to know how other languages/SDKs generally do this?

Reviewer 3: I think either is fine. We might want to look at the search data though. I know some symbols don't play well with search queries.

> This is the installation guide for the QDK environment to program hybrid Q# and Python programs.

Reviewer 3: Suggest changing this to more active language:

"Install the QDK for developing Q# operations called from a Python host program."

Reviewer 3: Or "Install the QDK to develop Python host programs to call Q# operations"

Author: Done in 467bf40

1. Pre-requisites

    - [Python](https://www.python.org/downloads/) 3.6 or later
    - The [PIP](https://pip.pypa.io/en/stable/installing) Python package manager
    - [.NET Core SDK 3.0 or later](https://www.microsoft.com/net/download)

1. Install the `qsharp` package, a Python package that allows the interoperation between Q# and Python.

    ```bash
    pip install qsharp
    ```

1. Install the `iqsharp`, an extension primarily used by Jupyter and Python that provides the core functionality for compiling and simulating Q# operations.

    ```bash
    dotnet tool install -g Microsoft.Quantum.IQSharp
    dotnet iqsharp install
    ```
  
1. Although it's not strictly necessary, we highly recommend to use the lightweight Visual Studio Code (VS Code) IDE to edit the Q# and Python files. Together with the QDK extension for VS Code this will provide code completion and syntax syntax highlighting features to make the coding easier.

    - Install [VS Code](https://code.visualstudio.com/download) (Windows, Linux and Mac)

>     - Install the [QDK extension for VS Code].

Author: Good catch

1. Verify the installation by creating a `Hello World` application

    - Create a minimal Q# operation, by creating a file called `Operation.qs`, and adding the following code to it:

        ```qsharp
        namespace HelloWorld
        {
            open Microsoft.Quantum.Intrinsic;
            open Microsoft.Quantum.Canon;

            operation SayHello() : Result {
                Message("Hello from quantum world!");
                return Zero;
            }
        }
        ```

    - Create a Python program called `hello_world.py` to call the Q# `SayHello()` operation:

        ```python
        import qsharp

        from HelloWorld import SayHello

        SayHello.simulate()
        ```

    - Run the program:

        ```bash
        python hello_world.py
        ```

    - Verify the output. Your program should output the following lines:

        ```bash
        Hello from quantum world!
       ```

> [!NOTE]

> > * You can also use Python Jupyter notebooks to write the classical Python program and call from the cells the Q# operations whenever they're needed. The Python code is just a normal Python program.

Reviewer 3: "whenever they're needed" isn't necessary here

Author: Done in 467bf40
